index.js

```diff
@@ -75,62 +75,55 @@ function startListening(httpServer) {
-
-            if (!isSupportedRequestType) {
-                // ignore for now
-                return
-            }
 
             let req = _.clone(abstractReq)
-            let res = {}
-            
             req.requestId = RequestId(data[0])
-            
             req.method = String(data[1]).toUpperCase()
-            req.url = res.path = data[2]
+            req.url = req.path = data[2]
             req.headers = data[3]
             req.body = data[4]
             req.params = []
-            
-            if (requestType !== supportedRequestTypes[0]) {
-                res = _.clone(abstractRes)
-
-                res.requestId = RequestId(data[0])
 
-                res.headers = {}
-                res.send = res.end = function send() {
-                    let statusCode = res.statusCode
-                    let body = arguments[0]
-                    if (arguments.length === 2) {
-                        statusCode = arguments[0]
-                        body = arguments[1]
-                    }
-
-                    let data = []
-
-                    data[0] = res.requestId
-                    data[1] = statusCode
-                    data[2] = res.headers
-                    data[3] = body
-
-                    connection.send(JSON.stringify(data))
-                }
-                res.setHeader = function(name, value) {
-                    res.headers[name] = value
-                }
-                res.unsetHeader = function(name) {
-                    res.headers[name] = undefined
+            let res = {}
+            res = _.clone(abstractRes)
+            res.requestId = RequestId(data[0])
+            res.headers = {}
+            res.send = res.end = function send() {
+                let statusCode = res.statusCode
+                let body = arguments[0]
+                if (arguments.length === 2) {
+                    statusCode = arguments[0]
+                    body = arguments[1]
                 }
+
+                connection.send(JSON.stringify([
+                    res.requestId,
+                    statusCode,
+                    res.headers,
+                    body,
+                ]));
+            }
+            res.setHeader = function(name, value) {
+                res.headers[name] = value
+            }
+            res.unsetHeader = function(name) {
+                res.headers[name] = undefined
             }
+
             if (requestType === supportedRequestTypes[0]) {
                 res.send = res.end = function() {
-                    console.warn('Response sednding not supported for NOTIFY')
+                    console.warn('Response sending not supported for NOTIFY')
                 }
             }
 
             console.info(colors.green(req.method + ':'), req.url)
-            app.handle(req, res)
+
+            if (supportedRequestTypes.indexOf(requestType) === -1) {
+                res.statusCode = 400;
```

Comment:
405 Method Not Allowed

Reply:
405 is not suitable because
> The method specified in the Request-Line is not allowed for the resource identified by the Request-URI.

but in our case method is not supported. I will use the 501 for this case